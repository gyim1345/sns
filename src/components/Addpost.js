import React, { useState } from "react";
import { addPostAPI } from "../apis/post";
import { useParams } from "react-router-dom";

function AddPost({ currentUser, posting, setPosting }) {
  const [input, setInput] = useState("");
  const { user } = useParams();

  const onChange = e => {
    setInput(e.target.value);
  };
  const onClick = async () => {
    try {
      const response = await addPostAPI(input, currentUser);
      setPosting([...posting, response]);
    } catch (e) {
      console.log(e);
    }
  };

  return (
    <>
      {user === currentUser && <input value={input} onChange={onChange} />}
      {user === currentUser && (
        <button type="button" onClick={onClick} id="buttonAdd">
          Add Post
        </button>
      )}
    </>
  );
}

// AddPost.propTypes = {
//   currentUser: PropTypes.string,
//   posting: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
//   setPosting: PropTypes.func
// };

// AddPost.defaultProps = {
//   currentUser: ""
// };

export default AddPost;
