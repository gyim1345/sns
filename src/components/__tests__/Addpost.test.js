import React from 'react';
import { shallow } from 'enzyme';
import Addpost from '../Addpost';
import { MemoryRouter } from 'react-router-dom';
import { currentUser } from './testjs';


describe('<Addpost />', () => {
  it('스냅샷 비교', () => {
    const wrapper = shallow(
      <MemoryRouter
        initialEntries={[{ pathname: '/gibong@gmail.com', key: 'testKey' }]}
      >
        <Addpost currentUser={currentUser} />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
