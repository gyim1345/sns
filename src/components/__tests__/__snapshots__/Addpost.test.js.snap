// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Addpost /> 스냅샷 비교 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/gibong@gmail.com",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/gibong@gmail.com",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <AddPost
    currentUser="gibong@gmail.com"
  />
</Router>
`;
